jina/executors/decorators.py

```diff
@@ -290,53 +288,80 @@ def arg_wrapper(*args, **kwargs):
+            #             )
+            #             b = new_memmap[slice_idx]
+            #             slice_idx = slice_idx[split_over_axis]
+            #             if slice_idx.start is None or slice_idx.stop is None:
+            #                 slice_idx = None
+            #             #elif yield_dict:
+            #             #    pass
+
+            print(f'yield_slice: {yield_slice}')
+            data_iterators = []
+            for i in range(0, slice_nargs):
+                data_iterators.append(
+                    batch_iterator(
+                        data[i],
+                        b_size,
+                        split_over_axis,
+                        yield_slice=yield_slice[i],
+                        # yield_dict=yield_dict[i],
+                    )
                 )
-                for i in range(0, slice_nargs)
-            ]
-
-            for new_args in zip(*data_iterators):
-                args[slice_on : slice_on + slice_nargs] = new_args  # a tuple of arrays
-                # print(f'data:{data}')
-                # print(f'args:{args}')
-                print(f'new arg: {new_args}')
-                slice_idx = new_args
                 '''
+                slice_idx = data[i]
+                print(f'slice_idx type :{type(slice_idx)}')
+                
                 new_memmap = np.memmap(
                     data.filename, dtype=data.dtype, mode='r', shape=data.shape
                 )
-                new_args = new_memmap[slice_idx]'''
+                new_args = new_memmap[slice_idx]
                 slice_idx = slice_idx[split_over_axis]
-                '''
+     
                 if slice_idx.start is None or slice_idx.stop is None:
-                    slice_idx = None'''
+                    slice_idx = None
                 if ordinal_idx_arg and slice_idx is not None:
-                    args[ordinal_idx_arg] = slice_idx
-                r = func(*args, **kwargs)
+                    args[ordinal_idx_arg] = slice_idx'''
+            copy_args = copy.copy(args)
+            for new_args in zip(*data_iterators):
+                new_args = list(new_args)
+                for i, (_yield_slice, _arg) in enumerate(zip(yield_slice, new_args)):
+                    if _yield_slice:
+                        slice_idx = _arg
+                        original_arg = args[slice_on + i]
+                        new_memmap = np.memmap(
+                            original_arg.filename,
+                            dtype=original_arg.dtype,
+                            mode='r',
+                            shape=original_arg.shape,
+                        )
+                        new_args[i] = new_memmap[slice_idx]
+                        slice_idx = slice_idx[split_over_axis]
+                        if slice_idx.start is None or slice_idx.stop is None:
+                            slice_idx = None
+                    # TODO: figure out what is ordinal_idx_arg
```

Comment:
see in the NmsLibIndexer how the ordinal_idx is used. What we can do is remove it from the core and have a custom batching implementation for that one.